src/inference/src/ie_core.cpp

```diff
@@ -1054,6 +1107,11 @@ class CoreImpl : public ie::ICore, public std::enable_shared_from_this<ie::ICore
 
         if (deviceName.empty()) {
             coreConfig.setAndUpdate(config);
+        } else {
+            auto cache_it = config.find(CONFIG_KEY(CACHE_DIR));
+            if (cache_it != config.end()) {
+                coreConfig.setCacheForDevice(cache_it->second, clearDeviceName);
+            }
```

Comment:
why do we have different methods for cases with and w/o device name? `setAndUpdate` removes `CONFIG_KEY(CACHE_DIR)` while `setCacheForDevice` does not

Reply:
Briefly, this is because now we have 'global cache_dir' and also devices can have separate 'cache_dir' per device. Setting 'global cache_dir' overwrites all existing cache_dir_per_device, while setting cache for for device does not.

setAndUpdate removes 'CACHE_DIR' because it is global config entry which is usually not passed to plugin (used by Core) and this setting can be retrieved from Core's CacheConfig.

But 'setCacheForDevice' - just stores this info in a map, and check is performed only on the time of loadNetwork, because on time of 'setCacheForDevice' plugin may not be loaded.